core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4799,6 +4808,57 @@ && isHourKeyword(qc.getAst().token) && qc.getAst().paramCount == 1 && qc.getAst(
         return root;
     }
 
+
+    private boolean aliasAppearsInColsAndColArgs(QueryModel model, CharSequence alias) {
+        boolean appearsInCols = false;
+        boolean appearsInArgs = false;
+        ObjList<QueryColumn> modelColumns = model.getColumns();
+        for (int i = 0; i < modelColumns.size(); i++) {
+            QueryColumn col = modelColumns.get(i);
+            if (col.getAst().type == 8) {
+                // function
+                ExpressionNode rhs = col.getAst().rhs;
+                ExpressionNode argNode = null;
+                while (rhs != null) {
+                    for (int j = 0; j < rhs.args.size(); j++) {
+                       argNode = rhs.args.getQuick(j);
+                       if (argNode == null) {
+                           continue;
+                       }
+                       if (argNode.token == alias) {
```

Comment:
One of `Chars#equals` methods should be used here. In Java, `argNode.token == alias` leads to reference comparison, not string comparison.